### PR TITLE
Machine wdt

### PR DESCRIPTION
**Pull request by ramya-subramanyam**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Added watchdog timer (WDT) machine module for psoc6 port.
functions implemented:

- Constructor
- feed()

### CODE REVIEW

File: tests/psoc6/wdt.py
Comment (NikhitaR-IFX) [inline tests/psoc6/wdt.py:1]: I think both tests can be inside hw_ext. Any problems you foresee to put it there?

**Review by NikhitaR-IFX — COMMENTED** [review]
We can try to put the tests into psoc6/hw_ext folder if there are no problems in doing so. Rest all looks good to me! Very neat 🚀

**Review by IFX-Anusha — APPROVED** [review]
Looks Good !👏 
But tests are failing right

File: tests/psoc6/wdt.py
Comment (jaenrig-ifx) [reply tests/psoc6/wdt.py:1]: Technically they don´t need any "hardware extension", why would we put them there?
It is true that they are a bit special, and not just callable any time as the rest of the psoc6/, neither as the hw_ext/ test. We still need to properly categorize and organize all these tests.

**Review by jaenrig-ifx — COMMENTED** [review]
I will run this in the HIL setup, and see what happens with the timing of the check_reset_cause test part.
But other than that, nothing to add. Very nice! Thanks @ramya-subramanyam !

File: tests/psoc6/wdt.py
Comment (NikhitaR-IFX) [reply tests/psoc6/wdt.py:1]: Yes that I agree is not requiring hardware extension.  My thought was we were moving the tests that were running on hardware from psoc6/ and eventually categorize like you mentioned. We can do that while we reorganize as well.

**Review by jaenrig-ifx — APPROVED** [review]

**Pull request merged**